migration-guide.md

```diff
@@ -0,0 +1,500 @@
+                                    CustomPOJO pojo_doc = OBJECT_MAPPER.treeToValue(document, CustomPOJO.class);
+                                    logger.info("----=>id: " + pojo_doc.getId());
+
+                                } catch (JsonProcessingException e) {
+                                    e.printStackTrace();
+                                }
+                            }
+                            logger.info("--->handleChanges() END");
+
+                        })
+                        .build();
+
+// ...
+
+ changeFeedProcessorInstance.start()
+                            .subscribeOn(Schedulers.elastic())
+                            .subscribe();
+```
+
+**Java SDK 3.x.x Async API:**
+
+```java
+ChangeFeedProcessor changeFeedProcessorInstance = 
+     ChangeFeedProcessor.Builder()
+                .hostName(hostName)
+                .feedContainer(feedContainer)
+                .leaseContainer(leaseContainer)
+                .handleChanges((List<CosmosItemProperties> docs) -> {
+                    logger.info("--->setHandleChanges() START");
+
+                    for (CosmosItemProperties document : docs) {
+                        try {
+
+                            // You are given the document as a CosmosItemProperties instance which you may
+                            // cast to the desired type.
+                            CustomPOJO pojo_doc = document.getObject(CustomPOJO.class);
+                            logger.info("----=>id: " + pojo_doc.id());
+
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                        }
+                    }
+                    logger.info("--->handleChanges() END");
+
+                })
+                .build();
+
+// ...
+
+ changeFeedProcessorInstance.start()
+                            .subscribeOn(Schedulers.elastic())
+                            .subscribe();
+```
+
+### Container TTL
+
+**Java SDK 4.0 Async API:**
+
+```java
+CosmosAsyncContainer container;
+
+// Create a new container with TTL enabled with default expiration value
+CosmosContainerProperties containerProperties = new CosmosContainerProperties("myContainer", "/myPartitionKey");
+containerProperties.setDefaultTimeToLiveInSeconds(90 * 60 * 60 * 24);
+container = database.createContainerIfNotExists(containerProperties, 400).block().getContainer();
+```
+
+**Java SDK 3.x.x Async API:**
+
+```java
+CosmosContainer container;
+
+// Create a new container with TTL enabled with default expiration value
+CosmosContainerProperties containerProperties = new CosmosContainerProperties("myContainer", "/myPartitionKey");
+containerProperties.defaultTimeToLive(90 * 60 * 60 * 24);
+container = database.createContainerIfNotExists(containerProperties, 400).block().container();
+```
+
+### Document TTL
+
+**Java SDK 4.0 Async API:**
+
+```java
+// Include a property that serializes to "ttl" in JSON
+public class SalesOrder
+{
```

Comment:
please be consistent on the java code style.
"{" on the same line as class definition.